docs/contribute/e2eTesting.md

```diff
@@ -1,66 +1,69 @@
- 
- * Have protractor installed (done in quickstart, by **npm install**)
-
- * Have the chrome webdriver installed (make sure your chromebrowser is up-to-date, > v30.x).
- This is done automatically if you use
- ```
- npm run pretest-e2e
- ```
- If there's a problem with the webdriver, try the following commands:
- 
-  On Unix systems:
-  ```
-  ./node_modules/protractor/bin/webdriver-manager update --chrome
-  ```
-  On Windows:
-  ```
-  node ./node_modules/protractor/bin/webdriver-manager update
-  ```
-  or
-  ```  
-  node node_modules\protractor\node_modules\webdriver-manager update
-  ```
-
-### Run the tests
-
-Open your console and change into the client directory.
+* Test data is in place and is not corrupt
+  (generate by running `gradlew clean build` in `scenarioo-docu-generation-example`).
 
-   ```
-   cd scenarioo-client
-   ```
+* Scenarioo Viewer is up and running:
+  * Scenarioo Viewer is built (run `gradlew clean build` in the project root directory).
+  * Scenarioo Viewer backend is running (deploy `scenarioo-latest.war` to Tomcat under the `/scenarioo` context,
+    IntelliJ can help with that).
+  * Scenarioo Viewer frontend is running (run using `npm start` in project root directory). 
 
-Serve the client:
+* Note: ChromeDriver will automatically be installed / updated when you run the E2E tests using the `npm run ...` 
+commands.
+
+
+### Run Tests
+
+Change into the client directory:
 
    ```
-   npm start
+   cd scenarioo-client
    ```
 
-Open a 2nd console and run the tests:
+Run the tests and also generate Scenarioo documentation (uses config file `protractor-e2e-scenarioo.conf.js`):
 
    ```
    npm run test-e2e-scenarioo
    ```
 
-If you do not want to generate a scenarioo docu from the tests, run this:
-
+Only run the tests, without generating Scenarioo documentation (uses config file `protractor-e2e.conf.js`):
    ```
    npm run test-e2e
    ```
 
-Npm runs the e2e tests according to the config file `protractor-e2e.conf.js` (no Scenarioo documentation) or `protractor-e2e-scenarioo.conf.js` (with Scenarioo documentation).
-
 This should open a new browser window, run all tests and log test-information to the console.
+
+
+### Debug Protractor Tests
+
+You can add either `browser.pause()` or `browser.debugger()` in your test code to set a breakpoint and start debugging
+from there. For details please see: http://blog.ng-book.com/debugging-protractor-tests/
+
+Please note that this approach does not work in nodejs 8.x anymore. See issue [676](https://github.com/scenarioo/scenarioo/issues/676).
+
+
+### Manually update ChromeDriver
+
+If there's a problem with ChromeDriver, try the following commands.
+ 
+On Linux:
+```
+./node_modules/protractor/bin/webdriver-manager update --chrome
+```
```

Comment:
If we have that problem often we should add that command to the npm scripts - such that you can use something like `npm run webdriver-manager update --chrome` or similar - somebody already added a webdriver-manager alias in package.json - not sure whether it works like that - but it should kind of.

Since we are using the chrome also for windows I wonder whether it wouldnt work as well for windows, so we could simply change the "pretest-e2e" script in package.json accordingly.